data_adapter_oemof/adapters.py

```diff
@@ -1,51 +1,281 @@
+                f"No busses found in facades fields for Dataadapter {self.__class__.__name__}"
+            )
+
+        bus_dict = {}
+        for bus in bus_occurrences_in_fields:  # emission_bus
+            # 1. Check for existing mappings
+            try:
+                bus_dict[bus] = self.bus_map[self.__class__.__name__][bus]
+                continue
+            except KeyError:
+                pass
+
+            # TODO: Make use of Parameter [stuct.csv]?
+            # Do we need parameter specific Bus structure? Maybe for multiple in/output?
+            if len(self.structure.keys()) == 1:
+                struct = list(self.structure.values())[0]
+            elif "default" in self.structure.keys():
+                struct = self.structure["default"]
+            else:
+                struct = self.structure
+                warnings.warn(
+                    "Please check structure and provide either one set of inputs/outputs "
+                    "or specify as default Parameter specific busses not implemented yet. "
+                    f"No Bus found for Process {self.process_name} in Adapter {self}"
+                )
+
+            # 2. Check for default busses
+            if bus in ("bus", "from_bus", "to_bus", "fuel_bus"):
+                if bus == "bus":
+                    busses = struct["inputs"] + struct["outputs"]
+                if bus in ("from_bus", "fuel_bus"):
+                    busses = struct["inputs"]
+                if bus == "to_bus":
+                    busses = struct["outputs"]
+                if len(busses) != 1:
+                    raise MappingError(
+                        f"Could not map {bus} to default bus - too many options"
+                    )
+                bus_dict[bus] = busses[0]
+                continue
+
+            # 3. Try to find close matches
+            match = difflib.get_close_matches(
+                bus,
+                struct["inputs"] + struct["outputs"],
+                n=1,
+                cutoff=0.2,
+            )[0]
+            if match:
+                bus_dict[bus] = match
+                continue
+
+            # 4. No mapping found
+            raise MappingError(
+                f"No Matching bus found for bus with `bus facade name` {bus}"
+                f" please adjust BUS_NAME_MAP or structure"
+            )
+
+        return bus_dict
+
+    def get_default_mappings(self):
+        """
+        :return: Dictionary for all fields that the facade can take and matching data
+        """
+
+        mapped_all_class_fields = {
+            field.name: value
+            for field in self.get_fields()
+            if (value := self.get(field.name, field.type)) is not None
+        }
+        mapped_all_class_fields.update(self.get_busses())
+        return mapped_all_class_fields
+
+    @staticmethod
+    def is_sequence(field_type: Type):
+        # TODO: Implement it using typing hints
+        return "Sequence" in str(field_type)
+
```

Comment:
Hard to review changes to mapping functionality but looking good and tests should catch major changes.